sp/src/game/server/hl2/npc_citizen17.cpp

```diff
@@ -839,6 +839,9 @@ void CNPC_Citizen::SelectModel()
 					}
 				}
 
+				// Models selected this way must be unique to avoid conflicts in save/restore
+				m_Type = CT_UNIQUE;
```

Comment:
I'm a little bit skeptical of this change. In Entropy : Zero 2, some citizen types have different effects such as jump rebels or rebel brutes. Using the SelectModel hook on them would remove these effects. I don't have a better alternative to suggest but I think maybe there should be a separate way to set m_Type or perhaps the save/restore bugs need to be addressed directly.

Reply:
I think this issue occurs because `CNPC_Citizen::SelectModel` is called from `Precache` and runs even after reloading a save. I'm not sure how to get around that safely.

Considering the `SelectModel` hook needs to be explicitly used by a VScript, this will not break E:Z2's custom citizen types unless they use a script that calls the hook and returns a new model. While I would want those types to continue using their unique abilities even if a different model is chosen, I don't think it's a big enough issue to block this PR.